core/common/src/rpc/web/WebAppRpcRequest.ts

```diff
@@ -34,7 +34,7 @@ export class WebAppRpcRequest extends RpcRequest {
   /** The maximum size permitted for an encoded component in a URL.
    * @note This is used for features like encoding the payload of a cacheable request in the URL.
    */
-  public static maxUrlComponentSize = 1024;
+  public static maxUrlComponentSize = 4096;
```

Comment:
In some testing with the itwin viewer and opening up views, I've already seen a mixed bag of GETs and POSTs for the same RPC operation. This is a direct result of the maxUrlComponentSize being too small for the parameters that are being passed to the backend. I'm unsure if its actually big enough, but if its not then all that will happen is the request will become a POST and everything should function normally. 

I'm not sure what negative effects there are from having larger fetch URLs, but I have added a test which uses the maxUrlComponentSize to pass a parameter as large as that max and asserts that it both remains a GET request and works as expected.

Reply:
Please consult @swbsi.

Reply:
We cannot safely change this to a large value. The maximum total url length is 2048 characters (in chrome...similar in other browsers). Our base urls can be rather long themselves...so we shouldn’t go much over 1024 here.

Reply:
But there is a difference between the url in the chrome browser and the url that we pass as part of a fetch, isn't there?

Reply:
The chrome limit is still active: https://chromium.googlesource.com/chromium/src/+/master/docs/security/url_display_guidelines/url_display_guidelines.md

It’s for security reasons (like denial of service attacks). It’s hard to believe there would be an exception for fetch-initiated requests.

Reply:
I assume you are referencing the "URL Length" section from that link?

![image](https://user-images.githubusercontent.com/22119573/160465222-bff17e61-e40a-4606-aab6-3de744a803a7.png)

"Chrome limits URLs to a maximum length of 2MB for practical reasons..." Isn't this much different than 2048 characters(I'm assuming 1 character = 1 byte)?

Reply:
@swbsi to quote the docs you linked to (emphasis added):

> In general, the web platform does not have limits on the length of URLs (although **2^31 is a common limit**). Chrome limits URLs to a **maximum length of 2MB** for practical reasons and to avoid causing denial-of-service problems in inter-process communication.

Setting aside the "2^31 is a common limit", 2MB is quite a bit larger than 2048 characters...  

Seems to me that as long as we don't care about supporting IE then 4096 should be _well_ within bounds.  I'd actually be concerned that it's still _too_ low...

@nick4598 do we have any telemetry that could inform whether 4096 is enough?   Also, have you looked into adding full-stack tests to confirm that RPC requests near our max length are handled correctly by the browser?

Reply:
I did add one full-stack-test which generates a string equal or close to the maxUrlComponentSize and asserted that it stays as a "GET". The RPC operation also returns that same string back to the caller, so it seems to work fine. 

As for whether 4096 is enough, we could probably use the current AppInsights stuff to look at customDimensions.requestData and figure out how large that could get for the RPC Operations we care about

Reply:
> I assume you are referencing the "URL Length" section from that link?
> 
> ![image](https://user-images.githubusercontent.com/22119573/160465222-bff17e61-e40a-4606-aab6-3de744a803a7.png)
> 
> "Chrome limits URLs to a maximum length of 2MB for practical reasons..." Isn't this much different than 2048 characters(I'm assuming 1 character = 1 byte)?

Yes it is :-)
I guess that was confirmation bias on my part. Or really fuzzy math...

I guess we are in the clear to change this limit then.

The other link (top google result) that says it’s 2048 is probably referring to the 32k display limit, I guess.